variable.c

```diff
@@ -2080,10 +2062,13 @@ rb_ivar_count(VALUE obj)
         return RCLASS_IV_COUNT(obj);
       default:
         if (FL_TEST(obj, FL_EXIVAR)) {
-            struct gen_ivtbl *ivtbl;
-
-            if (rb_gen_ivtbl_get(obj, 0, &ivtbl)) {
-                return gen_ivtbl_count(obj, ivtbl);
+            if (rb_shape_obj_too_complex(obj)) {
+                struct gen_ivtbl *ivtbl;
+                rb_gen_ivtbl_get(obj, 0, &ivtbl);
```

Comment:
If `rb_gen_ivtbl_get` fails, `ivtbl` will not be set, so dereferencing it will be undefined behavior.